core/src/main/kotlin/com/exactpro/th2/read/db/bootstrap/Main.kt

```diff
@@ -187,19 +195,47 @@ internal fun setupApp(
     READINESS_MONITOR.enable()
 }
 
+private fun createMessageLoader(
+    factory: CommonFactory,
+    componentBookName: String
+) = runCatching {
+    MessageSearcher(factory.grpcRouter.getService(DataProviderService::class.java)).run {
+        object : MessageLoader {
+            override fun load(dataSourceId: DataSourceId, properties: Map<String, String>): TableRow? =
+                searchLastOrNull(
+                    book = componentBookName,
+                    sessionAlias = dataSourceId.id,
+                    direction = com.exactpro.th2.common.grpc.Direction.FIRST,
+                    interval = Duration.ofDays(1), // FIXME: use parameter for that
```

Comment:
I am not sure that we need to expose a parameter to the user in this case. I think this can be left as is because the 1 day seems to be a good default and we cancel the request when we find the desired message